### PR TITLE
add Dockerfile + update docs

### PR DESCRIPTION
**Pull request by TheDen**
PR to add a layer-optimised alpine docker image. We've been using this `Dockerfile` internally with our CI for a few months now, and it's working great. It would be useful to have here for others to build and use (either to push to their own registries). 

Image tag is presupposed as `ambv/black`—ideally there would be a docker hub org `ambv` that and official image can be pushed to. But this is a good initial step.

Changes include:

* `./docker/build.sh` — builds the docker image `ambv/black:latest`
* `./docker/tag.sh` — tags `ambv/black:latest` as `ambv/black:<version>` where version is `__version__` from `black.py`. Not unlike how `setup.py`  grabs the verson, i.e., `_version_re = re.compile(r"__version__\s+=\s+(?P<version>.*)")` . However, I would prefer a nicer way to do this? Perhaps running `docker run -it ambv/black:latest --version` and parsing that?
* Docker image is 35.3MB (12.3MBs on docker registries after compression)
* Updated the `README.MD` to reflect the changes + add usage documentation.

### CODE REVIEW

**Comment by TheDen** [comment]
@ambv thoughts? Maybe this fell through the cracks

**Comment by nyurik** [comment]
I think this would be a great addition, thanks!

File: README.md [hunk 1]
```diff
@@ -759,6 +760,32 @@ affect your use case.
 
 This can be used for example with PyCharm's [File Watchers](https://www.jetbrains.com/help/pycharm/file-watchers.html).
 
+## Docker image
+
+### Building
+
+Running `build.sh` in the `docker/` folder will build the docker image tagged as `ambv/black:latest`
```
Comment (nvtkaszpir) [inline README.md]: todo: see if psf has official docker repo

File: README.md [hunk 2]
```diff
@@ -759,6 +760,32 @@ affect your use case.
...
+Running `tag.sh` in the `docker/` folder will tag the `ambv/black:latest` image based on the `__version__` in `black.py`
+
+### Usage
+
+`docker run ambv/black` defaults to the `--help` flag.
+
+To run a check on a python file:
+
+```bash
+docker run --read-only  -v $(pwd):$(pwd) -w $(pwd) ambv/black --check foo.py
```
Comment (nvtkaszpir) [inline README.md]: please use double quotes for elements such as `$(pwd)`

File: README.md [hunk 3]
```diff
@@ -759,6 +760,32 @@ affect your use case.
...
+```
+
+Note that the mounted directory has to be where the file exists, e.g.,
+
+```bash
+docker run --read-only  -v /home/ubunu/somefolder/:/black/ -w /black/ ambv/black --check foo.py
+```
+
```
Comment (nvtkaszpir) [inline README.md]: Maybe expand section with non-read-only parameters?

File: docker/build.sh [hunk 4]
```diff
@@ -0,0 +1,7 @@
+#!/bin/bash
+
+BASEDIR=$(dirname "$0")
+VERSION="latest"
```
Comment (nvtkaszpir) [inline docker/build.sh]: how about `VERSION=$(git describe --tags --dirty)`?

**Comment by kabirbaidhya** [comment]
I think this could be really a good addition. @TheDen are you still planning to work on this? 

Would be really great to have this merged. 
cc @nvtkaszpir

**Comment by TheDen** [comment]
@kabirbaidhya Yeah re-visiting it now. I think we should build the Docker image via the codebase, i.e., mount the repo and build `black`, that way it's tied to the commit. 

I'm trying to figure out how it's packaged, and will update this PR

**Comment by cooperlees** [comment]
Many thanks for the PR. The core black maintainers have decided that we think maintaining a black Docker imager out of scope for us.

Please feel free to offer this from your own repository and docker hub accounts.

**Pull request closed without merging**